src/dial.js

```diff
@@ -168,11 +168,9 @@ module.exports = function dial (swarm) {
           if (err) {
             return callback(err)
           }
-
-          pt.setReadable(conn)
-          pt.setWritable(conn)
-          pt.peerId = pi.id
-          callback(null, pt)
+          proxyConn.setInnerConn(conn)
+          proxyConn.peerId = pi.id
```

Comment:
but you could just change the impl on the connection so it is a setter, then you don't need to break anything:


```js
function Connection () {
  // ...
}

Object.setProperty(Connection.prototype, 'peerId', {
  set (id) {
    this._peerId = id
    // other things to do
  }
})


```